catalog/resource_catalog.go

```diff
@@ -57,6 +57,25 @@ func ResourceCatalog() *schema.Resource {
 		func(m map[string]*schema.Schema) map[string]*schema.Schema {
 			return m
 		})
+	update := func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
```

Comment:
Makes sense. I suggest you create an unexported package level utility "update factory" function of this. This way there's less repetition.